### ISSUE TITLE
Getting error while creating config map

### ISSUE THREAD

**Issue by rahulrajm** [hashicorp/learn-terraform-provision-eks-cluster#33]
terraform version 0.14
Getting Error: Unauthorized  while creating config map. Please fix

**Comment by im2nguyen** [comment]
Hello! Can you check if you have permission to create a config map? 

- https://aws.amazon.com/premiumsupport/knowledge-center/eks-api-server-unauthorized-error/
- https://github.com/terraform-aws-modules/terraform-aws-eks/issues/612

**Comment by rahulrajm** [comment]
I have admin privilege to create cluster but I am getting below error while
creating the cluster.

module.eks.kubernetes_config_map.aws_auth[0]: Creating...

Error: Unauthorized

**Comment by im2nguyen** [comment]
Can you try following the instructions for [You're the cluster creator](https://aws.amazon.com/premiumsupport/knowledge-center/eks-api-server-unauthorized-error/)

**Comment by rahulrajm** [comment]
The instructions are telling about error which we getting after cluster
creation.

I am hetting below error middle of the cluster creation

There is some issue in your code, please correct it or suggest me a stable
terraform code

**Comment by im2nguyen** [comment]
I haven't been able to replicate it, can you provide more information and the full error message?

**Comment by rahulrajm** [comment]
I have attached the error log. Please have a look.

**Comment by mijhael3000** [comment]
> terraform version 0.14
> Getting Error: Unauthorized while creating config map. Please fix

I solved this changing in my .aws/config file the region from us-west-2 to us-east-2

$ cat ~/.aws/config 
[default]
region = us-east-2

**Comment by rahulrajm** [comment]
@mijhael3000 After making the changes also, getting same error

**Comment by im2nguyen** [comment]
Can you create an issue in the [terraform-aws-eks](https://github.com/terraform-aws-modules/terraform-aws-eks) module repo? They should be able to help you

**Comment by siozosdr** [comment]
I'm getting the same error as well:
```bash
module.eks.kubernetes_config_map.aws_auth[0]: Creating...

Error: Unauthorized
```
I'm testing with `cluster_version="1.19"` instead of `"1.18"` but I don't think that it should matter.
Rest of the resources have been created successfully but not the config_map. If I try to access the EKS cluster from the AWS console I get the message:
```
Your current user or role does not have access to Kubernetes objects on this EKS cluster
This may be due to the current user or role not having Kubernetes RBAC permissions to describe cluster resources or not having an entry in the cluster’s auth config map
```
The AWS documentation only has to do with after creating the cluster which is NOT the case for me as well

**Comment by alanszlosek** [comment]
This should now be fixed by #64 and #65. Please let us know if you are still having issues.
